btrfs-progs: doesn't cross compile to `armv6l-linux` (from `x86_64-linux`)

### Describe the bug

Try to build `btrfs-progs` for `raspberryPi`.

Note, I'm trying staging because #168734 is necessary.

`nix build github:nixos/nixpkgs/staging#pkgsCross.raspberryPi.btrfs-progs` fails:

```
got build log for 'github:nixos/nixpkgs/staging#pkgsCross.raspberryPi.btrfs-progs' from 'daemon'
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/3brya9ik47l5qw6cndf7pf8cgjpymaaz-btrfs-progs-v5.16.2.tar.xz
source root is btrfs-progs-v5.16.2
setting SOURCE_DATE_EPOCH to timestamp 1645050425 of file btrfs-progs-v5.16.2/configure
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./config/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config/config.guess
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
configure flags: --disable-static --prefix=/nix/store/3zngz0dnlg3l31wgan0x83ndjicc5075-btrfs-progs-armv6l-unknown-linux-gnueabihf-5.16.2 --build=x86_64-unknown-linux-gnu --host=armv6l-unknown-linux-gnueabihf
checking for armv6l-unknown-linux-gnueabihf-gcc... armv6l-unknown-linux-gnueabihf-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether armv6l-unknown-linux-gnueabihf-gcc accepts -g... yes
checking for armv6l-unknown-linux-gnueabihf-gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... armv6l-unknown-linux-gnueabihf-gcc -E
checking for grep that handles long lines and -e... /nix/store/bqkx3pi50phcglv0l551jhp96bq8njl0-gnugrep-3.7/bin/grep
checking for egrep... /nix/store/bqkx3pi50phcglv0l551jhp96bq8njl0-gnugrep-3.7/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking for armv6l-unknown-linux-gnueabihf-gcc... (cached) armv6l-unknown-linux-gnueabihf-gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether armv6l-unknown-linux-gnueabihf-gcc accepts -g... (cached) yes
checking for armv6l-unknown-linux-gnueabihf-gcc option to accept ISO C89... (cached) none needed
checking whether C compiler accepts -std=gnu90... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... armv6l-unknown-linux-gnueabihf
checking for an ANSI C-conforming const... yes
checking for working volatile... yes
checking whether byte ordering is bigendian... no
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... 64
checking for a BSD-compatible install... /nix/store/hgl0ydlkgs6y6hx9h7k209shw3v7z77j-coreutils-9.0/bin/install -c
checking whether ln -s works... yes
checking for armv6l-unknown-linux-gnueabihf-ar... armv6l-unknown-linux-gnueabihf-ar
checking for rm... /nix/store/hgl0ydlkgs6y6hx9h7k209shw3v7z77j-coreutils-9.0/bin/rm
checking for rmdir... /nix/store/hgl0ydlkgs6y6hx9h7k209shw3v7z77j-coreutils-9.0/bin/rmdir
checking for openat... yes
checking for reallocarray... yes
checking for clock_gettime... yes
checking linux/perf_event.h usability... yes
checking linux/perf_event.h presence... yes
checking for linux/perf_event.h... yes
checking linux/hw_breakpoint.h usability... yes
checking linux/hw_breakpoint.h presence... yes
checking for linux/hw_breakpoint.h... yes
checking pkg-config is at least version 0.9.0... yes
checking execinfo.h usability... yes
checking execinfo.h presence... yes
checking for execinfo.h... yes
checking for backtrace... yes
checking for backtrace_symbols_fd... yes
checking for xmlto... /nix/store/aj13vfq0jp61an12dhz71alcb8py2wnw-xmlto-0.0.28/bin/xmlto
checking for mv... /nix/store/hgl0ydlkgs6y6hx9h7k209shw3v7z77j-coreutils-9.0/bin/mv
checking for a sed that does not truncate output... /nix/store/1442kn5q9ah0bhhqm99f8nr76diczqgm-gnused-4.8/bin/sed
checking for asciidoc... /nix/store/1hp17i8mcnwx537a0g1mfwxyf902pgy3-asciidoc-9.1.0/bin/asciidoc
checking for asciidoctor... no
checking for EXT2FS... yes
checking for COM_ERR... yes
checking for REISERFS... no
checking for FIEMAP_EXTENT_SHARED defined in linux/fiemap.h... yes
checking for EXT4_EPOCH_MASK defined in ext2fs/ext2_fs.h... yes
checking linux/blkzoned.h usability... yes
checking linux/blkzoned.h presence... yes
checking for linux/blkzoned.h... yes
checking for struct blk_zone.capacity... yes
checking for BLKGETZONESZ defined in linux/blkzoned.h... yes
checking for BLKID... yes
checking for UUID... yes
checking for ZLIB... yes
checking for ZSTD... yes
checking for LIBUDEV... yes
checking for a Python interpreter with version >= 3.4... python
checking for python... /nix/store/hym1n0ygqp9wcm7pxn4sfrql3fg7xa09-python3-3.9.12/bin/python
checking for python version... 3.9
checking for python platform... linux
checking for GNU default python prefix... ${prefix}
checking for GNU default python exec_prefix... ${exec_prefix}
checking for python script directory (pythondir)... ${PYTHON_PREFIX}/lib/python3.9/site-packages
checking for python extension module directory (pyexecdir)... ${PYTHON_EXEC_PREFIX}/lib/python3.9/site-packages
checking for PYTHON... yes
checking for lzo_version in -llzo2... yes
configure: creating ./config.status
config.status: creating Makefile.inc
config.status: creating Documentation/Makefile
config.status: creating version.h
config.status: creating libbtrfsutil/libbtrfsutil.pc
config.status: creating config.h

	btrfs-progs:        v5.16.2 
	libbtrfs:           0.1.2 (deprecated)
	libbtrfsutil:       1.2.0

	prefix:             /nix/store/3zngz0dnlg3l31wgan0x83ndjicc5075-btrfs-progs-armv6l-unknown-linux-gnueabihf-5.16.2
	exec prefix:        ${prefix}

	bindir:             ${exec_prefix}/bin
	libdir:             ${exec_prefix}/lib
	includedir:         ${prefix}/include
	pkgconfigdir:       ${libdir}/pkgconfig

	compiler:           armv6l-unknown-linux-gnueabihf-gcc
	CFLAGS:             -g -O2 -Wall -D_FORTIFY_SOURCE=2
	LDFLAGS:            

	programs:           yes
	shared libraries:   yes
	static libraries:   no
	documentation:      yes
	doc generator:      asciidoc
	backtrace support:  yes
	btrfs-convert:      yes (ext2)
	btrfs-restore zstd: yes
	Python bindings:    yes
	Python interpreter: /nix/store/hym1n0ygqp9wcm7pxn4sfrql3fg7xa09-python3-3.9.12/bin/python
	crypto provider:    builtin 
	libudev:            yes
	zoned device:       yes
	
	Type 'make' to compile.

@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: -j12 -l12 SHELL=/nix/store/fcd0m68c331j7nkdxvnnpb8ggwsaiqac-bash-5.1-p16/bin/bash udevruledir=\$\(out\)/lib/udev/rules.d
    [CC]     btrfs.o
    [CC]     kernel-lib/list_sort.o
    [CC]     kernel-lib/radix-tree.o
    [CC]     kernel-lib/raid56.o
    [CC]     kernel-lib/rbtree.o
    [CC]     kernel-lib/tables.o
    [CC]     kernel-shared/backref.o
    [CC]     kernel-shared/extent_io.o
    [CC]     kernel-shared/file-item.o
    [CC]     kernel-shared/file.o
    [CC]     kernel-shared/free-space-cache.o
    [CC]     kernel-shared/free-space-tree.o
    [CC]     kernel-shared/inode-item.o
    [CC]     kernel-shared/inode.o
    [CC]     kernel-shared/print-tree.o
    [CC]     kernel-shared/root-tree.o
    [CC]     kernel-shared/extent-tree.o
    [CC]     kernel-shared/disk-io.o
    [CC]     kernel-shared/dir-item.o
In file included from [01m[K/nix/store/35pjnvb806b8q94kvjngqh2crav07294-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/string.h:519[m[K,
                 from [01m[K./kerncompat.h:30[m[K,
                 from [01m[Kkernel-shared/print-tree.c:23[m[K:
In function '[01m[Kstrncpy[m[K',
    inlined from '[01m[Kbg_flags_to_str[m[K' at [01m[Kkernel-shared/print-tree.c:197:4[m[K:
[01m[K/nix/store/35pjnvb806b8q94kvjngqh2crav07294-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/bits/string_fortified.h:95:10:[m[K [01;35m[Kwarning: [m[K'[01m[K__builtin_strncpy[m[K' specified bound 64 equals destination size [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation-Wstringop-truncation]8;;[m[K]
   95 |   return [01;35m[K__builtin___strncpy_chk (__dest, __src, __len,[m[K
      |          [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
   96 | [01;35m[K      __glibc_objsize (__dest))[m[K;
      |       [01;35m[K~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
    [CC]     kernel-shared/delayed-ref.o
    [CC]     kernel-shared/ulist.o
    [CC]     kernel-shared/uuid-tree.o
    [CC]     kernel-shared/transaction.o
    [CC]     kernel-shared/zoned.o
    [CC]     common/device-scan.o
    [CC]     common/device-utils.o
    [CC]     common/extent-cache.o
    [CC]     common/format-output.o
    [CC]     common/parse-utils.o
    [CC]     common/open-utils.o
    [CC]     common/path-utils.o
    [CC]     common/rbtree-utils.o
    [CC]     common/repair.o
    [CC]     common/send-stream.o
    [CC]     common/send-utils.o
    [CC]     common/string-table.o
    [CC]     common/task-utils.o
    [CC]     common/units.o
    [CC]     common/utils.o
    [CC]     common/messages.o
    [CC]     cmds/receive-dump.o
    [CC]     check/qgroup-verify.o
    [CC]     common/help.o
    [CC]     crypto/xxhash.o
In file included from [01m[K/nix/store/35pjnvb806b8q94kvjngqh2crav07294-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/stdio.h:888[m[K,
                 from [01m[Kcommon/help.c:17[m[K:
In function '[01m[Kfprintf[m[K',
    inlined from '[01m[Kusage_command_internal[m[K' at [01m[Kcommon/help.c:257:3[m[K,
    inlined from '[01m[Kusage_command_usagestr[m[K' at [01m[Kcommon/help.c:271:8[m[K,
    inlined from '[01m[Kusage[m[K' at [01m[Kcommon/help.c:331:2[m[K:
[01m[K/nix/store/35pjnvb806b8q94kvjngqh2crav07294-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/bits/stdio2.h:105:10:[m[K [01;35m[Kwarning: [m[K'[01m[K%s[m[K' directive argument is null [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wformat-overflow=-Wformat-overflow=]8;;[m[K]
  105 |   return [01;35m[K__fprintf_chk (__stream, __USE_FORTIFY_LEVEL - 1, __fmt,[m[K
      |          [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
  106 | [01;35m[K   __va_arg_pack ())[m[K;
      |    [01;35m[K~~~~~~~~~~~~~~~~~[m[K
In function '[01m[Kfprintf[m[K',
    inlined from '[01m[Kusage_command_internal[m[K' at [01m[Kcommon/help.c:254:3[m[K,
    inlined from '[01m[Kusage_command_usagestr[m[K' at [01m[Kcommon/help.c:271:8[m[K,
    inlined from '[01m[Kusage[m[K' at [01m[Kcommon/help.c:331:2[m[K:
[01m[K/nix/store/35pjnvb806b8q94kvjngqh2crav07294-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/bits/stdio2.h:105:10:[m[K [01;35m[Kwarning: [m[K'[01m[K%s[m[K' directive argument is null [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wformat-overflow=-Wformat-overflow=]8;;[m[K]
  105 |   return [01;35m[K__fprintf_chk (__stream, __USE_FORTIFY_LEVEL - 1, __fmt,[m[K
      |          [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
  106 | [01;35m[K   __va_arg_pack ())[m[K;
      |    [01;35m[K~~~~~~~~~~~~~~~~~[m[K
    [CC]     crypto/sha224-256.o
    [CC]     crypto/hash.o
    [CC]     libbtrfsutil/stubs.o
    [CC]     libbtrfsutil/subvolume.o
    [CC]     cmds/subvolume.o
    [CC]     cmds/subvolume-list.o
    [CC]     cmds/filesystem.o
    [CC]     cmds/device.o
    [CC]     cmds/scrub.o
    [CC]     cmds/inspect.o
    [CC]     cmds/balance.o
    [CC]     cmds/send.o
    [CC]     cmds/receive.o
[01m[Kcmds/inspect.c:[m[K In function '[01m[K__ino_to_path_fd[m[K':
[01m[Kcmds/inspect.c:70:21:[m[K [01;35m[Kwarning: [m[Karray subscript '[01m[Ki[m[K' is outside the bounds of an interior zero-length array '[01m[K__u64[0][m[K' {aka '[01m[Klong long unsigned int[][m[K'} [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wzero-length-bounds-Wzero-length-bounds]8;;[m[K]
   70 |   ptr += [01;35m[Kfspath->val[i][m[K;
      |          [01;35m[K~~~~~~~~~~~^~~[m[K
In file included from [01m[Kcmds/inspect.c:27[m[K:
[01m[K./ioctl.h:500:8:[m[K [01;36m[Knote: [m[Kwhile referencing '[01m[Kval[m[K'
  500 |  __u64 [01;36m[Kval[m[K[0];  /* out */
      |        [01;36m[K^~~[m[K
[01m[Kcmds/inspect.c:[m[K In function '[01m[Kcmd_inspect_logical_resolve[m[K':
[01m[Kcmds/inspect.c:247:13:[m[K [01;35m[Kwarning: [m[Karray subscript -1 is outside array bounds of '[01m[Kchar[4096][m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Warray-bounds-Warray-bounds]8;;[m[K]
  247 |     [01;35m[Kpath_ptr[-1][m[K = '\0';
      |     [01;35m[K~~~~~~~~^~~~[m[K
[01m[Kcmds/inspect.c:156:7:[m[K [01;36m[Knote: [m[Kwhile referencing '[01m[Kfull_path[m[K'
  156 |  char [01;36m[Kfull_path[m[K[PATH_MAX];
      |       [01;36m[K^~~~~~~~~[m[K
In file included from [01m[K/nix/store/35pjnvb806b8q94kvjngqh2crav07294-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/string.h:519[m[K,
                 from [01m[K./kerncompat.h:30[m[K,
                 from [01m[Kcmds/inspect.c:26[m[K:
In function '[01m[Kstrncpy[m[K',
    inlined from '[01m[Kcmd_inspect_logical_resolve[m[K' at [01m[Kcmds/inspect.c:282:5[m[K:
[01m[K/nix/store/35pjnvb806b8q94kvjngqh2crav07294-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/bits/string_fortified.h:95:10:[m[K [01;35m[Kwarning: [m[K'[01m[K__builtin_strncpy[m[K' specified bound 4096 equals destination size [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation-Wstringop-truncation]8;;[m[K]
   95 |   return [01;35m[K__builtin___strncpy_chk (__dest, __src, __len,[m[K
      |          [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
   96 | [01;35m[K      __glibc_objsize (__dest))[m[K;
      |       [01;35m[K~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
    [CC]     cmds/quota.o
    [CC]     crypto/blake2b-ref.o
In file included from [01m[K/nix/store/35pjnvb806b8q94kvjngqh2crav07294-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/string.h:519[m[K,
                 from [01m[K./kerncompat.h:30[m[K,
                 from [01m[Kcmds/scrub.c:19[m[K:
In function '[01m[Kstrncpy[m[K',
    inlined from '[01m[Kscrub_start[m[K' at [01m[Kcmds/scrub.c:1363:3[m[K:
[01m[K/nix/store/35pjnvb806b8q94kvjngqh2crav07294-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/bits/string_fortified.h:95:10:[m[K [01;35m[Kwarning: [m[K'[01m[K__builtin_strncpy[m[K' output may be truncated copying 107 bytes from a string of length 4095 [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation-Wstringop-truncation]8;;[m[K]
   95 |   return [01;35m[K__builtin___strncpy_chk (__dest, __src, __len,[m[K
      |          [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
   96 | [01;35m[K      __glibc_objsize (__dest))[m[K;
      |       [01;35m[K~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
    [CC]     cmds/replace.o
    [CC]     cmds/rescue.o
    [CC]     cmds/rescue-chunk-recover.o
    [CC]     cmds/rescue-super-recover.o
    [CC]     cmds/property.o
    [CC]     cmds/filesystem-usage.o
    [CC]     cmds/inspect-dump-tree.o
    [CC]     cmds/restore.o
    [CC]     cmds/inspect-dump-super.o
    [CC]     cmds/inspect-tree-stats.o
    [CC]     check/main.o
    [CC]     mkfs/common.o
    [CC]     check/mode-common.o
    [CC]     check/mode-lowmem.o
    [CC]     libbtrfsutil/errors.o
    [CC]     libbtrfsutil/filesystem.o
    [CC]     libbtrfsutil/qgroup.o
    [CC]     cmds/filesystem-du.o
    [CC]     mkfs/rootdir.o
    [CC]     btrfs-map-logical.o
    [CC]     image/main.o
    [CC]     image/sanitize.o
    [CC]     convert/common.o
    [CC]     convert/source-fs.o
    [CC]     convert/source-ext2.o
    [CC]     convert/source-reiserfs.o
    [CC]     btrfs-corrupt-block.o
In file included from [01m[K/nix/store/35pjnvb806b8q94kvjngqh2crav07294-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/string.h:519[m[K,
                 from [01m[Kkerncompat.h:30[m[K,
                 from [01m[Kbtrfs-corrupt-block.c:26[m[K:
In function '[01m[Kstrncpy[m[K',
    inlined from '[01m[Kmain[m[K' at [01m[Kbtrfs-corrupt-block.c:1318:5[m[K:
[01m[K/nix/store/35pjnvb806b8q94kvjngqh2crav07294-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/bits/string_fortified.h:95:10:[m[K [01;35m[Kwarning: [m[K'[01m[K__builtin_strncpy[m[K' specified bound 80 equals destination size [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation-Wstringop-truncation]8;;[m[K]
   95 |   return [01;35m[K__builtin___strncpy_chk (__dest, __src, __len,[m[K
      |          [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
   96 | [01;35m[K      __glibc_objsize (__dest))[m[K;
      |       [01;35m[K~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
    [CC]     libbtrfs/send-stream.o
    [CC]     libbtrfs/send-utils.o
    [LD]     libbtrfsutil.so.1.2.0
Making all in Documentation
    [ASCII]  btrfs-ioctl.xml
    [ASCII]  btrfs-man5.xml
    [XMLTO]  btrfs-ioctl.3
    [XMLTO]  btrfs-man5.5
rm btrfs-man5.xml btrfs-ioctl.xml
    [AR]     libbtrfsutil.a
    [CC]     convert/main.o
    [CC]     btrfs-select-super.o
    [LN]     libbtrfsutil.so.1
    [LN]     libbtrfsutil.so
    [CC]     btrfstune.o
    [PY]     libbtrfsutil
    [CC]     btrfs-find-root.o
    [CC]     mkfs/main.o
In file included from [01m[K/nix/store/35pjnvb806b8q94kvjngqh2crav07294-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/string.h:519[m[K,
                 from [01m[K./kerncompat.h:30[m[K,
                 from [01m[Kconvert/main.c:83[m[K:
In function '[01m[Kstrncpy[m[K',
    inlined from '[01m[Kmain[m[K' at [01m[Kconvert/main.c:1940:6[m[K:
[01m[K/nix/store/35pjnvb806b8q94kvjngqh2crav07294-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/bits/string_fortified.h:95:10:[m[K [01;35m[Kwarning: [m[K'[01m[K__builtin_strncpy[m[K' specified bound 37 equals destination size [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation-Wstringop-truncation]8;;[m[K]
   95 |   return [01;35m[K__builtin___strncpy_chk (__dest, __src, __len,[m[K
      |          [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
   96 | [01;35m[K      __glibc_objsize (__dest))[m[K;
      |       [01;35m[K~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
    [CC]     cmds/qgroup.o
In file included from [01m[K/nix/store/hym1n0ygqp9wcm7pxn4sfrql3fg7xa09-python3-3.9.12/include/python3.9/Python.h:50[m[K,
                 from [01m[Kbtrfsutilpy.h:27[m[K,
                 from [01m[Kerror.c:20[m[K:
[01m[K/nix/store/hym1n0ygqp9wcm7pxn4sfrql3fg7xa09-python3-3.9.12/include/python3.9/pyport.h:741:2:[m[K [01;31m[Kerror: [m[K#error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
  741 | #[01;31m[Kerror[m[K "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
      |  [01;31m[K^~~~~[m[K
error: command '/nix/store/lxmy4hfcpg59knmh5iw37kmwx6f45ps2-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-wrapper-10.3.0/bin/armv6l-unknown-linux-gnueabihf-gcc' failed with exit code 1
    [CC]     crypto/crc32c.o
make: *** [Makefile:562: libbtrfsutil_python] Error 1
make: *** Waiting for unfinished jobs....

```

staging is at : 9bdcee8f0451a95ad846eb064a29179467969b64

--- a/pkgs/development/python-modules/setuptools/default.nix
+++ b/pkgs/development/python-modules/setuptools/default.nix
@@ -1,6 +1,7 @@
 { stdenv
 , buildPythonPackage
 , fetchFromGitHub
+, fetchpatch
 , python
 , bootstrapped-pip
 , lib
@@ -27,6 +28,14 @@ let
     patches = [
       ./tag-date.patch
       ./setuptools-distutils-C++.patch
+      # Use sysconfigdata to find headers. Fixes cross-compilation of extension modules.
+      # https://github.com/pypa/distutils/pull/145
+      (fetchpatch {
+        url = "https://github.com/pypa/distutils/commit/aed7294b7b0c228cc0666a8b04f2959bf310ab57.patch";
+        hash = "sha256-/9+TKv0nllBfnj48zcXLrOgyBj52dBIVbrpnIaQ4O84=";
+        stripLen = 2;
+        extraPrefix = "setuptools/_distutils/";
+      })
     ];
 
     buildPhase = ''

